src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TryStatement.cs

```diff
@@ -16,13 +16,20 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
             var origSawAwait = _sawAwait;
             _sawAwait = false;
 
-            ImmutableArray<BoundCatchBlock> catchBlocks = (ImmutableArray<BoundCatchBlock>)this.VisitList(node.CatchBlocks);
+            // When optimizing and we have an empty try block, we can discard the catch blocks.
+            // In theory we could do better by detecting *effectively* empty catch blocks, but this is the most common case.
+            ImmutableArray<BoundCatchBlock> catchBlocks =
+                node.TryBlock.Statements.Length == 0 && this._compilation.Options.OptimizationLevel == OptimizationLevel.Release
+                    ? ImmutableArray<BoundCatchBlock>.Empty
+                    : (ImmutableArray<BoundCatchBlock>)this.VisitList(node.CatchBlocks);
```

Comment:
Semantic errors are reported in earlier passes. The only errors we need to report here are problems that prevent us from generating code. If we can drop the code then there are no problems.